raiden/smart_contracts/ChannelManagerLibrary.sol

```diff
@@ -105,23 +130,49 @@ library ChannelManagerLibrary {
         partner_channels.length--;
 
         // remove address from all_channels
-        for (uint k = 0; k < self.all_channels.length; ++k) {
-            if (self.all_channels[k] == channel_address) {
-                self.all_channels[k] == self.all_channels[self.all_channels.length - 1];
-                self.all_channels.length--;
-                break;
-            }
-        }
+        uint i = getIndex(self, channel_address);
+        self.all_channels[i] == self.all_channels[self.all_channels.length - 1];
+        self.all_channels.length--;
+
+        // remove channel from channel_addresses. 
+        self.channel_addresses[partyHash(caller_address, partner)] = 0x0;
 
         self.node_channels[caller_address] = our_channels;
         self.node_channels[partner] = partner_channels;
     }
 
+    /// @notice getIndex(address) to get the index of a specific channel address
```

Comment:
As discussed please remove the type from the docstrings in the entire PR since solidity is a typed language and it is redundant information.